daemon/cmd/daemon_main.go

```diff
@@ -1621,7 +1549,39 @@ func (d *Daemon) initKVStore() {
 	}
 }
 
-func daemonModule(ctx context.Context, cleaner *daemonCleanup, shutdowner fx.Shutdowner) (*Daemon, error) {
+// registerDaemonHooks registers the lifecycle hooks for the part of the cilium-agent that has
+// not yet been modularized.
+//
+// The Daemon struct and objects referred to from there are not supplied to the graph as
+// object creation in Daemon is intertwined with side-effectful initialization. This stops
+// us from constructing and inspecting the dependency graph, so instead the daemon is constructed
+// and started via an OnStart hook.
+//
+// If an object still owned by Daemon is required in a module, it should be provided indirectly, e.g. via
+// a callback.
+func registerDaemonHooks(lc fx.Lifecycle, shutdowner fx.Shutdowner) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	cleaner := NewDaemonCleanup()
+	lc.Append(fx.Hook{
+		OnStart: func(context.Context) error {
+			go runDaemon(ctx, cleaner, shutdowner)
```

Comment:
why the `go`?

Reply:
Start hooks cannot block as they're executed sequentially. If you prefer, I could rename this to `startDaemon` and do a `go srv.Serve()` from there instead?

Reply:
I think a comment in the code is sufficient